File: app/src/main/java/com/example/myapplication/ViewModels/TemplateViewModel.java
package com.example.myapplication.ViewModels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.myapplication.Models.Template;
import com.example.myapplication.Repositories.TemplateRepository;

import java.util.List;

public class TemplateViewModel extends AndroidViewModel {

    private TemplateRepository repository;
    private LiveData<List<Template>> allTemplates;

    public TemplateViewModel(@NonNull Application application) {
        super(application);

        repository = new TemplateRepository(application);
        allTemplates = repository.getAllTemplates();
    }

    public void insert(Template template) {
        repository.insert(template);
    }

    public void update(Template template) {
        repository.update(template);
    }

    public void delete(Template template) {
        repository.delete(template);
    }

    public LiveData<List<Template>> getAllTemplatest() {
        return allTemplates;
    }

    public Template getTemplateById(int id) { return repository.getTemplateById(id); }
}

File: app/src/main/java/com/example/myapplication/ViewModels/TemplateViewModel.kt
package com.example.myapplication.ViewModels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.example.myapplication.Models.Template
import com.example.myapplication.Repositories.TemplateRepository
import java.util.concurrent.ExecutionException

class TemplateViewModel(application: Application) : AndroidViewModel(application) {
    private val repository: TemplateRepository
    val allTemplatest: LiveData<List<Template>>
    fun insert(template: Template?) {
        repository.insert(template)
    }

    fun update(template: Template?) {
        repository.update(template)
    }

    fun delete(template: Template?) {
        repository.delete(template)
    }

    @Throws(ExecutionException::class, InterruptedException::class)
    fun getTemplateById(id: Int): Template {
        return repository.getTemplateById(id)
    }

    init {
        repository = TemplateRepository(application)
        allTemplatest = repository.allTemplates
    }
}